x/pki/handler_test.go

```diff
@@ -135,1703 +135,1983 @@ func Setup(t *testing.T) *TestSetup {
-
-	// propose x509 leaf certificate as root
-	proposeAddX509RootCert := types.NewMsgProposeAddX509RootCert(setup.Trustee1.String(), testconstants.LeafCertPem, testconstants.Info, testconstants.Vid)
-	_, err := setup.Handler(setup.Ctx, proposeAddX509RootCert)
-	require.Error(t, err)
-	require.True(t, pkitypes.ErrInappropriateCertificateType.Is(err))
-}
-
-func TestHandler_ProposeAddX509RootCert_ProposedCertificateAlreadyExists(t *testing.T) {
-	setup := Setup(t)
-
-	// propose adding of x509 root certificate
-	proposeAddX509RootCert := types.NewMsgProposeAddX509RootCert(setup.Trustee1.String(), testconstants.RootCertPem, testconstants.Info, testconstants.Vid)
-	_, err := setup.Handler(setup.Ctx, proposeAddX509RootCert)
-	require.NoError(t, err)
-
-	// store another account
-	anotherAccount := GenerateAccAddress()
-	setup.AddAccount(anotherAccount, []dclauthtypes.AccountRole{dclauthtypes.Trustee}, 1)
-
-	// propose adding of the same x509 root certificate again
-	proposeAddX509RootCert = types.NewMsgProposeAddX509RootCert(anotherAccount.String(), testconstants.RootCertPem, testconstants.Info, testconstants.Vid)
-	_, err = setup.Handler(setup.Ctx, proposeAddX509RootCert)
-	require.Error(t, err)
-	require.True(t, pkitypes.ErrProposedCertificateAlreadyExists.Is(err))
-}
-
-func TestHandler_ProposeAddX509RootCert_CertificateAlreadyExists(t *testing.T) {
-	setup := Setup(t)
-
-	// store x509 root certificate
-	rootCertificate := rootCertificate(testconstants.Address1)
-	setup.Keeper.SetUniqueCertificate(
-		setup.Ctx,
-		uniqueCertificate(rootCertificate.Subject, rootCertificate.SerialNumber),
-	)
-	setup.Keeper.AddApprovedCertificate(setup.Ctx, rootCertificate)
-
-	// propose adding of the same x509 root certificate
-	proposeAddX509RootCert := types.NewMsgProposeAddX509RootCert(setup.Trustee1.String(), testconstants.RootCertPem, testconstants.Info, testconstants.Vid)
-	_, err := setup.Handler(setup.Ctx, proposeAddX509RootCert)
-	require.Error(t, err)
-	require.True(t, pkitypes.ErrCertificateAlreadyExists.Is(err))
+	setup.AddAccount(accAddress, []dclauthtypes.AccountRole{role}, vid)
+	return accAddress
 }
 
-func TestHandler_ProposeAddX509RootCert_ForDifferentSerialNumber(t *testing.T) {
-	setup := Setup(t)
-
-	// store root certificate with different serial number
-	rootCertificate := rootCertificate(setup.Trustee1)
-	rootCertificate.SerialNumber = SerialNumber
-	setup.Keeper.SetUniqueCertificate(
-		setup.Ctx,
-		uniqueCertificate(rootCertificate.Subject, rootCertificate.SerialNumber),
-	)
-	setup.Keeper.AddApprovedCertificate(setup.Ctx, rootCertificate)
-
-	// propose second root certificate
-	proposeAddX509RootCert := types.NewMsgProposeAddX509RootCert(setup.Trustee1.String(), testconstants.RootCertPem, testconstants.Info, testconstants.Vid)
+// Creates a proposal to add the X509 root certificate
+func ProposeX509RootCert(setup *TestSetup, trusteeAddr sdk.AccAddress, rootCertPem, info string, vid int32) (*types.MsgProposeAddX509RootCert, error) {
```

Comment:
It's better to rename `rootCertPem` to just `cert` because it may be at least not a root certificate.

Reply:
Done